### PR TITLE
fix: Implement V3 event handling to getRefund logic in BundleDataClient 

### PR DESCRIPTION
**Pull request by nicholaspai**
- Allows InventoryClient to better project upcoming refunds
- Allows reporter to do the same

### CODE REVIEW

File: test/utils/utils.ts [hunk 1]
```diff
@@ -805,25 +806,10 @@ export function getDefaultBlockRange(toBlockOffset: number): number[][] {
...
-): Record<
-  string,
-  {
-    refunds: Record<string, BigNumber>;
-    fills: Fill[];
-    totalRefundAmount: BigNumber;
-    realizedLpFees: BigNumber;
-  }
-> {
+export function createRefunds(address: string, refundAmount: BigNumber, token: string): CombinedRefunds {
```
Comment (james-a-morris) [inline test/utils/utils.ts:809]: This function was largely simplified - do we still need it or can we build this `CombinedRefunds` inlined in the code?

File: src/clients/InventoryClient.ts [hunk 2]
```diff
@@ -230,6 +231,7 @@ export class InventoryClient {
       // Consider any refunds from executed and to-be executed bundles.
       totalRefundsPerChain = await this.getBundleRefunds(l1Token);
     } catch (e) {
+      this.log("Failed to get bundle refunds, defaulting refund chain to hub chain");
```
Comment (james-a-morris) [inline src/clients/InventoryClient.ts:234]: Do we have access to a logger here?

File: src/clients/BundleDataClient.ts [hunk 3]
```diff
@@ -217,18 +224,24 @@ export class BundleDataClient {
...
     // Refunds that will be processed in the next bundle that will be proposed after the current pending bundle
     // (if any) has been fully executed.
-    return (await this.loadData(futureBundleEvaluationBlockRanges, this.spokePoolClients, false)).fillsToRefund;
+    const { fillsToRefund, bundleFillsV3, expiredDepositsToRefundV3 } = await this.loadData(
+      futureBundleEvaluationBlockRanges,
+      this.spokePoolClients,
+      false
+    );
+    const combinedRefunds = getRefundsFromBundle(bundleFillsV3, fillsToRefund, expiredDepositsToRefundV3);
+    return combinedRefunds;
```
Comment (james-a-morris) [inline src/clients/BundleDataClient.ts:233]:
```suggestion
    return getRefundsFromBundle(bundleFillsV3, fillsToRefund, expiredDepositsToRefundV3);
```

**Review by james-a-morris — COMMENTED** [review]
Initial pass. I will pull this PR down and leave more substantial comments now

File: test/utils/utils.ts (hunk 1 above)
Comment (james-a-morris) [reply test/utils/utils.ts:809]: eh - this is a test file so there's not much pressure to switch this. My above comment is only actionable if you deem it worthwhile.

File: src/dataworker/DataworkerUtils.ts [hunk 4]
```diff
@@ -293,23 +293,19 @@ function buildV3SlowFillLeaf(deposit: interfaces.V3Deposit): V3SlowFillLeaf {
   };
 }
 
-export function _buildRelayerRefundRoot(
```
Comment (james-a-morris) [inline src/dataworker/DataworkerUtils.ts:296]: OOC - why did we shift this function definition down in the new diff?

File: src/dataworker/DataworkerUtils.ts [hunk 5]
```diff
@@ -293,23 +293,19 @@ function buildV3SlowFillLeaf(deposit: interfaces.V3Deposit): V3SlowFillLeaf {
...
+  [repaymentChainId: number]: {
+    [repaymentToken: string]: interfaces.Refund;
+  };
+};
 
-  // Create a combined `refunds` object containing refunds for V2 + V3 fills
-  // and expired deposits.
+// Create a combined `refunds` object containing refunds for V2 + V3 fills
+// and expired deposits.
+export function getRefundsFromBundle(
```
Comment (james-a-morris) [inline src/dataworker/DataworkerUtils.ts:304]: Ah I see - this function has been split up. +1 for the reusability

File: src/clients/BundleDataClient.ts [hunk 6]
```diff
@@ -428,12 +441,14 @@ export class BundleDataClient {
         spokePoolClients
       );
       this.bundleTimestampCache = bundleBlockTimestamps;
-      this.logger.debug({
-        at: "BundleDataClient#loadData",
-        message: "Bundle block timestamps",
-        bundleBlockTimestamps,
-        blockRangesForChains,
-      });
+      if (logData) {
```
Comment (james-a-morris) [inline src/clients/BundleDataClient.ts:443]: +1 good call

File: src/clients/InventoryClient.ts (hunk 2 above)
Comment (nicholaspai) [reply src/clients/InventoryClient.ts:234]: yeah look below

**Review by james-a-morris — COMMENTED** [review]
I really like this PR. Splitting up the code so we can use it elsewhere was a good move. 

I'll preemptively approve so that I'm not a blocker since it's critical. But I'd recommend Matt also takes a look.

**Review by mrice32 — APPROVED** [review]
LGTM!

**Pull request merged**